Sources/ConcurrencyHelpers/ConcurrentMap.swift

```diff
@@ -0,0 +1,84 @@
+extension RandomAccessCollection where Self: Sendable, Element: Sendable {
+    func concurrentMap<B: Sendable>(minBatchSize: Int = 4096, _ transform: @Sendable @escaping (Element) async -> B) async -> [B] {
+        precondition(minBatchSize >= 1)
+        let n = self.count
+        let batchCount = (n + minBatchSize - 1) / minBatchSize
+        if batchCount < 2 {
+            return await withTaskGroup(of: B.self) { group in
+                var results: [B] = []
+                for element in self {
+                    group.addTask { await transform(element) }
+                }
+                for await result in group {
+                    results.append(result)
+                }
+                return results
+            }
+        }
+
```

Comment:
Can we create the final array in one allocation and then fill it concurrently?

Reply:
Then we get issues with @Sendable, didn't find any way to convince the compiler yet that we are ok.

Reply:
Seems need to play with unsafe mutable buffers as we cannot initialize allocated memory. [](edited-with-Axolo)